pkg/sqlite/scene.go

```diff
@@ -609,56 +615,20 @@ func sceneMoviesCriterionHandler(qb *sceneQueryBuilder, movies *models.MultiCrit
 	return h.handler(movies)
 }
 
-func scenePerformerTagsCriterionHandler(qb *sceneQueryBuilder, performerTagsFilter *models.MultiCriterionInput) criterionHandlerFunc {
+func scenePerformerTagsCriterionHandler(qb *sceneQueryBuilder, tags *models.HierarchicalMultiCriterionInput) criterionHandlerFunc {
 	return func(f *filterBuilder) {
-		if performerTagsFilter != nil && len(performerTagsFilter.Value) > 0 {
-			qb.performersRepository().join(f, "performers_join", "scenes.id")
-			f.addJoin("performers_tags", "performer_tags_join", "performers_join.performer_id = performer_tags_join.performer_id")
+		if tags != nil && len(tags.Value) > 0 {
+			addHierarchicalWithClause(f, tags.Value, "tags_for_performer", tagTable, "tags_relations", "", tags.Depth)
 
-			var args []interface{}
-			for _, tagID := range performerTagsFilter.Value {
-				args = append(args, tagID)
-			}
+			f.addWith(`performer_tags AS (
+SELECT ps.scene_id, t.id AS root_tag_id FROM performers_scenes ps
+INNER JOIN performers_tags pt ON pt.performer_id = ps.performer_id
+INNER JOIN tags_for_performer t ON t.child_id = pt.tag_id
+)`)
 
-			if performerTagsFilter.Modifier == models.CriterionModifierIncludes {
-				// includes any of the provided ids
-				f.addWhere("performer_tags_join.tag_id IN "+getInBinding(len(performerTagsFilter.Value)), args...)
-			} else if performerTagsFilter.Modifier == models.CriterionModifierIncludesAll {
-				// includes all of the provided ids
-				f.addWhere("performer_tags_join.tag_id IN "+getInBinding(len(performerTagsFilter.Value)), args...)
-				f.addHaving(fmt.Sprintf("count(distinct performer_tags_join.tag_id) IS %d", len(performerTagsFilter.Value)))
-			} else if performerTagsFilter.Modifier == models.CriterionModifierExcludes {
-				f.addWhere(fmt.Sprintf(`not exists
-					(select performers_scenes.performer_id from performers_scenes
-						left join performers_tags on performers_tags.performer_id = performers_scenes.performer_id where
-						performers_scenes.scene_id = scenes.id AND
-						performers_tags.tag_id in %s)`, getInBinding(len(performerTagsFilter.Value))), args...)
-			}
-		}
-	}
-}
-
-func handleScenePerformerTagsCriterion(query *queryBuilder, performerTagsFilter *models.MultiCriterionInput) {
```

Comment:
@WithoutPants This method was added by you when you added support for performer tags, but to me it seems like it has been unused all the time, can you confirm this? (Obviously code still compiles fine, but I don't know whether it can be invoked dynamically by means of reflection or something alike).